docs/library_development.md

```diff
@@ -0,0 +1,19 @@
+
+Creating a Vinum library
+==============================
+ 
+Follow these steps to create a Vinum library:
+
+ 1. Include the `library_api.h`.
```

Comment:
I don't think the library developer should need to include a header from the stdlib, the developer should only need the header from vinumc